### ISSUE TITLE
blobfuse fails to mount on RHEL/CentOS 7 without debug (-d) or foreground (-f) FUSE flags

### ISSUE THREAD

**Issue by stewartadam** [Azure/azure-storage-fuse#518]
### Which version of the blobfuse was used?
1.3.4

### Which OS (please include version) are you using?
CentOS Linux release 7.8.2003 (Core)

### What problem was encountered?
blobfuse mounts and lists files correctly on RHEL7 when enabling FUSE debug (`-d`) or foreground (`-f`), but fails when those flags aren't present; so if I had to guess the issues folks are seeing on CentOS 7 might be a race condition, not necessarily due to curl <7.35 [as discussed in #272](https://github.com/Azure/azure-storage-fuse/issues/272#issuecomment-707192058) (at least not for RHEL7's version of curl 7.29.0).

### Have you found a mitigation/solution?
No

### By default, blobfuse logs errors to syslog. If this is relevant, is there anything in the syslog that might be helpful?
Heres's an excerpt of the log when `-d` or `-f` CLI arguments are present:
```
...
Oct 12 16:35:07 linuxvm blobfuse[3340]: Function azs_readdir, in file /usr/pipeline/blobfuse/azure-storage-fuse/blobfuse/directoryapis.cpp, line 133: azs_readdir : About to call list_blobs.  Container = blobfuse-test, delimiter = /, continuation = , prefix =
Oct 12 16:35:07 linuxvm blobfuse[3340]:  ==> REQUEST/RESPONSE :: GET https://accountName.blob.core.windows.net/blobfuse-test?comp=list&delimiter=/&include=metadata&maxresults=5000&restype=container&sv=2019-12-12&ss=btqf&srt=sco&st=2020-10-12T16%3A01%3A29Z&se=2020-11-13T17%3A01%3A00Z&sp=rl&sig=REDACTED?&User-Agent=azure-storage-fuse/1.3.4&x-ms-date=Mon, 12 Oct 2020 16:35:07 GMT&x-ms-version=2018-11-09&Transfer-Encoding=--------------------------------------------------------------------------------RESPONSE Status :: 200 :: REQ ID : 4575c20d-901e-0126-65b5-a0ec82000000
Oct 12 16:35:07 linuxvm blobfuse[3340]: Function azs_readdir, in file /usr/pipeline/blobfuse/azure-storage-fuse/blobfuse/directoryapis.cpp, line 205: #### So far 3 items retreived in 1 iterations.
...
```

And without them / mounting normally:
```
...
Oct 12 16:35:45 linuxvm blobfuse[3365]: Function azs_readdir, in file /usr/pipeline/blobfuse/azure-storage-fuse/blobfuse/directoryapis.cpp, line 133: azs_readdir : About to call list_blobs.  Container = blobfuse-test, delimiter = /, continuation = , prefix =
Oct 12 16:35:45 linuxvm blobfuse[3365]:  ==> REQUEST/RESPONSE :: GET https://accountName.blob.core.windows.net/blobfuse-test?comp=list&delimiter=/&include=metadata&maxresults=5000&restype=container&sv=2019-12-12&ss=btqf&srt=sco&st=2020-10-12T16%3A01%3A29Z&se=2020-11-13T17%3A01%3A00Z&sp=rl&sig=REDACTED?&User-Agent=azure-storage-fuse/1.3.4&x-ms-date=Mon, 12 Oct 2020 16:35:45 GMT&x-ms-version=2018-11-09&Transfer-Encoding=--------------------------------------------------------------------------------RESPONSE Status :: 0 ::
...
```

**Comment by pmoreira1** [comment]
I found exactly the same issue. 
@stewartadam Have you been able to find a work around?

**Comment by stewartadam** [comment]
Nothing without debugging/foreground flags :/

**Comment by NaraVen** [comment]
@pmoreira1 , @stewartadam . This was the workaround to get blobfuse working on RHEL versions below 8. This is caused due to an SSL handshake issue with Curl in the earlier versions. We have to fork our codebase to release for versions below RHEL 8. Instead of forkign we recommend that you use -f or -d while mounting blobfuse.

**Comment by pmoreira1** [comment]
I'm actually running curl 7.73.0, wouldn't that solve the handshake issue?

**Comment by NaraVen** [comment]
@pmoreira1 , @stewartadam  When I upgrade Curl to 7.54 I get it working without the -d or -f. Can you both try the below steps.
yum install gnutls gnutls-devel libssh2 libssh2-devel
yum install gcc
wget https://curl.haxx.se/download/curl-7.54.0.tar.gz
sudo tar -xvzf curl-7.54.0.tar.gz && cd curl-7.54.0/
sudo ./configure --with-gnutls --with-libssh2 --prefix=/usr/local
make
make install
sudo rm /usr/bin/curl
sudo ln -s /usr/local/bin/curl /usr/bin/curl

**Comment by stewartadam** [comment]
I will give this a try when I have a moment on a test machine, but to be clear for others having this issue **do not to this on any production machines**. I would not consider this a solution, just a verification that older curls do not work with current blobfuse.

Overwriting your system curl will introduce several issues:
1. If a vulnerability is discovered in curl 7.54 your machine will remain unpatched and vulnerable
1. If a system (RHEL/CentOS) update for curl is released, this local installation of 7.54 may get overwritten
1. This may break compatibility other installed programs depending on curl

If curl 7.54 is required, then blobfuse RPM needs to require that version specifically and should consider providing+maintaining that package in the yum repos alongside blobfuse if RHEL7 is still supported.

**Comment by vibhansa-msft** [comment]
Blobfuse version 1.3.6 is compatible with older cURL version and mount works on RHEL7 and CentOS7. Kindly upgrade using below link.
https://github.com/Azure/azure-storage-fuse/releases/tag/blobfuse-1.3.6
